test/pjrt/test_experimental_pjrt.py

```diff
@@ -52,6 +50,49 @@ def test_xla_device_error(self):
     with self.assertRaises(IndexError):
       xm.xla_device(10)
 
+  @staticmethod
+  def _pjrt_default_device(env_vars: Dict[str, Optional[str]]):
+    for k, v in env_vars.items():
+      if v:
+        os.environ[k] = v
+      else:
+        os.environ.pop(v)
+
+    return pjrt.device_type()
+
+  @parameterized.named_parameters(('default', {}, True), ('no_default', {
+      'PJRT_SELECT_DEFAULT_DEVICE': '0'
+  }, False), ('pjrt_cpu', {
+      'PJRT_DEVICE': 'CPU',
+      'PJRT_SELECT_DEFAULT_DEVICE': '0'
+  }, True), ('xrt_tpu', {
+      'XRT_TPU_CONFIG': 'localservice;0;localhost:51011'
+  }, False), ('pjrt_tpu_precedence', {
+      'PJRT_DEVICE': 'TPU',
+      'XRT_TPU_CONFIG': 'localservice;0;localhost:51011'
+  }, True), ('xrt_gpu', {
+      'GPU_NUM_DEVICES': '4'
+  }, False), ('pjrt_gpu', {
+      'PJRT_DEVICE': 'GPU',
+      'GPU_NUM_DEVICES': '4'
+  }, True))
```

Comment:
should we add tests for xla_dist case, for example `XRT_LOCAL_WORKER` in https://github.com/pytorch/xla/blob/master/third_party/xla_client/computation_client.cc#L196. It is a bit tricky since even if we set all env vars, it will fail because there is no XRT server to distribue the TPU topology in this case.. 

Maybe we should just have a test with mesh server setup and make sure it doesn't got overwrite to PJRT_DEVICE. I think we can configure that mesh timeout to some shorter time, defualt is 300 s.

Reply:
or we can just set any `XRT_` and expect the test to fail instead of getting a PJRT device.

Reply:
Good idea. I don't know why I forgot to add a unit test for the bug I'm fixing.

Added an an `xla_dist` test case that just sets `XRT_LOCAL_WORKER`. Since we don't actually initialize the client in this unit test, it works without having to set up a mesh server.